internal/terraform/node_resource_abstract_instance.go

```diff
@@ -663,7 +669,7 @@ func (n *NodeAbstractResourceInstance) refresh(ctx EvalContext, deposedKey state
 		panic("new state is cty.NilVal")
 	}
 
-	if !resp.NewState.IsWhollyKnown() {
+	if !resp.NewState.IsWhollyKnown() && !ctx.Deferrals().DeferralAllowed() {
```

Comment:
nit: quick comment explaining why it's okay for things to be unknown during a deferral